src/database/DataTypes/TransactionBatch.js

```diff
@@ -169,6 +169,11 @@ export class TransactionBatch extends Realm.Object {
     return `${this.itemBatch} in a ${this.transaction.type}`;
   }
 
+  get transactionItem() {
+    const { transaction } = this;
+    return transaction.items.find(({ batches }) => batches.some(batch => batch.id === this.id));
```

Comment:
I think this works:
transaction.items.filtered('batches.id = $0', this.id)

Reply:
couldn't get that query to work. Would be wicked if it did though